src/apify_client/clients/resource_clients/task.py

```diff
@@ -0,0 +1,214 @@
+            timeout_secs: (int, optional): Optional timeout for the run, in seconds. By default, the run uses timeout specified in the task settings.
+            task_input (dict, optional): Task input dictionary
+
+        Returns:
+            The updated task
+        """
+        new_fields = {
+            "name": name,
+            "options": {
+                "build": build,
+                "memoryMbytes": memory_mb,
+                "timeoutSecs": timeout_secs,
+            },
+            "input": task_input,
+        }
+
+        return self._update(_filter_out_none_values_recursively(new_fields))
+
+    def delete(self) -> None:
+        """Delete the task.
+
+        https://docs.apify.com/api/v2#/reference/actor-tasks/task-object/delete-task
+        """
+        return self._delete()
+
+    def start(
+        self,
+        *,
+        task_input: Optional[Dict[str, Any]] = None,
+        build: Optional[str] = None,
+        memory_mb: Optional[int] = None,
+        timeout_secs: Optional[int] = None,
+        wait_for_finish: Optional[int] = None,
+        webhooks: Optional[List[Dict]] = None,
+    ) -> Dict:
+        """Start a task and immediately return the Run object.
+
+        https://docs.apify.com/api/v2#/reference/actor-tasks/run-collection/run-task
+
+        Args:
+            task_input (dict, optional): Task input dictionary
+            build (str, optional): Specifies the actor build to run. It can be either a build tag or build number.
+                                   By default, the run uses the build specified in the task settings (typically latest).
+            memory_mb (int, optional): Memory limit for the run, in megabytes. By default, the run uses a memory limit specified in the task settings.
+            timeout_secs: (int, optional): Optional timeout for the run, in seconds. By default, the run uses timeout specified in the task settings.
+            wait_for_finish: (bool, optional): The maximum number of seconds the server waits for the run to finish.
+                                               By default, it is 0, the maximum value is 300.
+            webhooks (list, optional): Specifies optional webhooks associated with the actor run, which can be used to receive a notification
+                                       e.g. when the actor finished or failed. Note: if you already have a webhook set up for the actor or task,
+                                       you do not have to add it again here.
+
+        Returns:
+            The run object
+        """
+        request_params = self._params(
+            build=build,
+            memory=memory_mb,
+            timeout=timeout_secs,
+            waitForFinish=wait_for_finish,
+            webhooks=_encode_json_to_base64(webhooks) if webhooks is not None else [],
+        )
+
+        response = self.http_client.call(
```

Comment:
Not sure, but shouldn't this rather be "not finished yet" too? Seems everything in this function should instead be one call to the RunClient, or we will repeat the same code several times.

Reply:
JS client does it explicitly (https://github.com/apify/apify-client-js/blob/master/src/resource_clients/task.js#L78-L94), I've been mimicking that.

Reply:
Sure, but that doesn't mean it can't be improved on.

```
         response = self.http_client.call(
            url=self._url('runs'),
            method='POST',
            headers={'content-type': 'application/json; charset=utf-8'},
            json=task_input,
            params=request_params,
        )

        return _parse_date_fields(_pluck_data(response.json()))
```
This could be a single short function call instead of one long one and one short one. And that's not even mentioning that if we ever decide to change the way we parse the response for instance, we would have to go through all these files.

Reply:
It would be easy to create such a function, but it would need to accept many arguments (at least `url`, `method`, `json`, `params`, `data`) and would execute 2 lines of code. And also, what would we call that function? `_call_get_data_and_parse_dates`?  I can't think of a good short name :D.

I'm not convinced this would help us, it feels more like passing around arguments. But I'm open for an another opinion,  @fnesveda, @mnmkng ? Thx

Reply:
_execute_request(...) ?